examples/token-sale/token_sale.go

```diff
@@ -82,108 +87,90 @@ func getIntFromDB(ctx storage.Context, key []byte) int {
 	return res
 }
 
-// InCirculation return the amount of total tokens that are in circulation.
-func (t TokenConfig) InCirculation(ctx storage.Context) int {
-	return getIntFromDB(ctx, t.CirculationKey)
+// InCirculation returns the amount of total tokens that are in circulation.
+func InCirculation() int {
+	return getIntFromDB(ctx, token.CirculationKey)
 }
 
 // AddToCirculation sets the given amount as "in circulation" in the storage.
-func (t TokenConfig) AddToCirculation(ctx storage.Context, amount int) bool {
-	supply := getIntFromDB(ctx, t.CirculationKey)
+func AddToCirculation(amount int) bool {
+	supply := getIntFromDB(ctx, token.CirculationKey)
 	supply += amount
-	storage.Put(ctx, t.CirculationKey, supply)
+	storage.Put(ctx, token.CirculationKey, supply)
 	return true
 }
 
-// TokenSaleAvailableAmount returns the total amount of available tokens left
+// AvailableAmount returns the total amount of available tokens left
 // to be distributed.
-func (t TokenConfig) TokenSaleAvailableAmount(ctx storage.Context) int {
-	inCirc := getIntFromDB(ctx, t.CirculationKey)
-	return t.TotalSupply - inCirc
+func AvailableAmount() int {
+	inCirc := getIntFromDB(ctx, token.CirculationKey)
+	return token.TotalSupply - inCirc
 }
 
-// Main smart contract entry point.
-func Main(operation string, args []interface{}) interface{} {
-	var (
-		trigger = runtime.GetTrigger()
-		cfg     = NewTokenConfig()
-		ctx     = storage.GetContext()
-	)
+// init initializes runtime trigger, TokenConfig and storage context before any
+// other contract method is called
+func init() {
+	trigger = runtime.GetTrigger()
+	token = newTokenConfig()
+	ctx = storage.GetContext()
+}
 
+// CheckWitness checks whether the invoker is the owner of the contract.
+func CheckWitness() bool {
 	// This is used to verify if a transfer of system assets (NEO and Gas)
 	// involving this contract's address can proceed.
 	if trigger == runtime.Verification {
```

Comment:
For some reason I think that this one actually has to be done with `runtime.Application` trigger to prevent bad behavior. So it's not a separate method, but rather a part of some methods that should only be run by the owner.

Reply:
I fixed trigger, made this function unexported and used by `Approve(...)`. Is there something else that should be done?

Reply:
It's good enough now, I think.